### PR TITLE
Making search results clickable (part of #450)

### PR DESCRIPTION
**Pull request by brismuth**
Making search results clickable (part of #450)
Adding isReadOnly to tabs
Adding optional callback to openFile

### CODE REVIEW

File: js/ui/projectManager.js [hunk 1]
```diff
@@ -344,7 +344,7 @@ define([
       });
     },
     
-    openFile: function(path) {
+    openFile: function(path, c) {
```
Comment (brismuth) [inline js/ui/projectManager.js]: I needed to add this callback so I could jump to the line number after opening.

File: js/ui/searchbar.js [hunk 2]
```diff
@@ -145,19 +160,25 @@ define([
...
           return c();
         }
 
-        var printResult = function(index, str) {
-          self.appendToResults(self.formatResultCode(index, str));
-        };
+        var file = new File(nodeEntry);
 
         file.read(function(err, data) {
+          if (!options.running) {
```
Comment (brismuth) [inline js/ui/searchbar.js]: this was also needed in order to make sure short circuiting cuts off at the right number.

**Comment by thomaswilburn** [comment]
Let me think about it a little. I didn't really want to add any universal event listeners just for search tabs, but if it's the best way, we'll go with it.

**Comment by brismuth** [comment]
Yeah, I understand. Another way we could do it would be to add a class to
the editor for search tabs only, then attach a click listener to that
class, but bound to the parent so we don't have to rebind it every time.

**Comment by thomaswilburn** [comment]
Still think I'd rather have this using the marker system, the more I think about it. I'd be okay adding event delegation to the editor for the marker class, but not I think for the search tabs. Give me a little while, and I'll tackle this on my own, this has just been a very busy week.

The SearchHighlight dynamic marker that Ace uses internally is helpful: https://github.com/ajaxorg/ace/blob/36e6744a5f40df0da52ff22b3bc729657c056e09/lib/ace/search_highlight.js

**Comment by thomaswilburn** [comment]
Also, just being picky, but I prefer not to use `isSomething` in item attributes. So for example, the read-only flag is probably a good one to have on the Tab class, but I'd rather have `readOnly` to `isReadOnly`, the latter just seems like extra verbiage to me.

**Comment by brismuth** [comment]
Fair enough. I'll leave the branch how it is for now in case using the marker API doesn't work out for whatever reason. It really does work very well, not sure if you had the chance to try it out.

If it's okay with you I'll open another pull request with some of the changes that we want to keep, like the callback from projects, the fix to short circuiting in search, and the small organizational changes to code in searchbar.js.

**Comment by thomaswilburn** [comment]
Sure, cleanup pull requests are always welcome. And if I can't get the 
marker thing working, I'll definitely come back to this.

**Pull request closed without merging**